source-code/core/src/lint/rules/brandingRule.ts

```diff
@@ -0,0 +1,24 @@
+import type { Context } from '../context.js'
+import { createRule } from '../rule.js'
+
+export const createBrandingRule = (brandName: string, incorrectBrandingNames: string[]) => createRule('inlang.brandingRule', () => {
+	let context: Context
+
+	return {
+		initialize: (config) => {
+			context = config.context
+		},
+		visitors: {
+			Pattern: ({ target }) => {
+				if (!target) return
```

Comment:
Why is "skip" not returned as in the other rules?

Reply:
Because here we also want to check the reference resource for errors. For the other rules, checking the reference does not make sense.

Reply:
Ah, is returning `undefined` like "continue with linting the other nodes" but don't "skip linting other rules"?  

```ts
if (!target) return 

// is "equivalent" to

if (!target) return "continue"
```

Reply:
yes, if
 - `'skip'` get's returned, then children will not be processed
 - `void` or `undefined` get's returned, then chidren will be preocessed with the payload that get's passed by the parent
 - somevalue get's returned, children will be process with that payload

Reply:
This pattern is called "early return" and prevents a nesting level in the implementation.
```ts
function doSomething() {
	if (condition) {
		console.log('Hello')
	}
}
```

vs.

```ts
function doSomething() {
	if (!condition) return

	console.log('Hello')
}
```

When you have a few lines of code inside the `if` statement, then the second variant is easier to read. And once you have multiple `if`s then you would end up with multiple levels of nesting. Therefore I prefer early returns.
_Except for single if statements where the condition is not trivial_